src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressException.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Diagnostics.Monitoring.Extension.Common
+{
+    /// <summary>
+    /// Exception that egress providers can throw when an operational error occurs (e.g. failed to write the stream data).
+    /// </summary>
+    internal sealed class EgressException : MonitoringException
```

Comment:
Do we want to have a separate exception hierarchy here? It seems unnecessary to share a base exception type since we can't catch out of proc exceptions anyway.

Reply:
I think that it would still be good for us to have a single exception type that can be thrown that indicates that we know something is wrong (e.g. validation of the options) as opposed to some logical error (e.g. NRE). But this can definitely be flattened to one type.